internal/daemonset/daemonset.go

```diff
@@ -195,6 +200,25 @@ func (dc *daemonSetGenerator) SetDriverContainerAsDesired(ctx context.Context, d
 		Selector: &metav1.LabelSelector{MatchLabels: standardLabels},
 	}
 
+	if fw := mod.Spec.ModuleLoader.Container.Modprobe.FirmwarePath; fw != "" {
+		firmwareVolume := v1.Volume{
+			Name: nodeVarLibFirmwareVolumeName,
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: nodeVarLibFirmwarePath,
+					Type: &hostPathDirectoryOrCreate,
+				},
+			},
+		}
+		ds.Spec.Template.Spec.Volumes = append(ds.Spec.Template.Spec.Volumes, firmwareVolume)
+
+		firmwareVolumeMount := v1.VolumeMount{
+			Name:      nodeVarLibFirmwareVolumeName,
+			MountPath: nodeVarLibFirmwarePath,
+		}
+		ds.Spec.Template.Spec.Containers[0].VolumeMounts = append(ds.Spec.Template.Spec.Containers[0].VolumeMounts, firmwareVolumeMount)
```

Comment:
I think it would probably be safer to have `ds.Spec.Template.Spec.Containers[0]` as a variable, to add the volume mount to it and to ultimately add it to the `TemplateSpec`.

Reply:
I've also split the volumes to follow the same pattern. Is that what you had in mind?

Reply:
Yes, although I would have put that work above the assigment of `ds.Spec` line 124.

Reply:
Done.